### ISSUE TITLE
node:crypto is broken on 0.5.0

### ISSUE THREAD

**Issue by pateketrueke** [oven-sh/bun#1839]
### What version of Bun is running?

0.5.0

### What platform is your computer?

Darwin 20.6.0 x86_64 i386

### What steps can reproduce the bug?

@robobun

```js
import crypto from 'node:crypto';

function fn() {
  crypto.createHash('sha1').update(Math.random(), 'ascii').digest('base64');
}

for (let i = 0; i < 10; i++) fn();

console.log('OK');
```

### What is the expected behavior?

No error, `OK` should be printed.

### What do you see instead?

> fish: Job 1, 'bun run broke.js' terminated by signal SIGSEGV (Address boundary error)

### Additional information

![SCR-20230118-wi6](https://user-images.githubusercontent.com/206371/213362220-3bb87bab-0509-460c-b0a9-06393b33e81a.png)

**Comment by robobun** [comment]
@pateketrueke here you go!
```tsx

```
<details>
  <summary>Code</summary>
  
```tsx
import crypto from 'node:crypto';

function fn() {
  crypto.createHash('sha1').update(Math.random(), 'ascii').digest('base64');
}

for (let i = 0; i < 10; i++) fn();

console.log('OK');
```
</details>

_Ran using the latest build of Bun, an all-in-one JavaScript runtime._

**Comment by Jarred-Sumner** [comment]
I think we are not coercing to string but need to be

**Comment by Jarred-Sumner** [comment]
and the segfault is happening because the code for type checking is likely returning the 0 value to JS instead of throwing an exception

**Comment by Jarred-Sumner** [comment]
Fixed in 4c0b0e2e8ee3a2710ea3d5b3711efc905f6f2be5
